### ISSUE TITLE
[Bug] Debounce of input (i.e. change:data event) leads to small probability of dirty reads

### ISSUE THREAD

**Issue by darylteo** [ckeditor/ckeditor5-vue#95]
Because we have to rely on `@input` event for getting the data from ckeditor, there is a very small probability that when an action is performed within 1s of the last content change, the event will not have fired before this action is performed, and the action performed uses outdated information.

I don't really know how to solve this. I have used the `@ready` event to get access to the editor, and have tried using `editor.model.document.differ` but each time `isEmpty` is `true`.

Right now my proposed workaround is to actively block an action being performed by using `@keyup` set a timestamp and to defer actions for a small period after this timestamp. But, again, any insight that you can provide which may make this easier to solve would be handy.

I suspect this debounce might stem from the core engine, so if this is the wrong place to discuss please let me know and I can move the issue.

Thanks very much!

**Comment by Mgsy** [comment]
Hi, thanks for the report. I can confirm it and we are aware of this issue. 

cc @oleq

**Comment by oleq** [comment]
I understand your concerns, @darylteo.

We debounce the `@input` because calling `editor.getData()` comes at a cost. Especially when there's a lot of content, many subsequent `editor.getData()` [would kill the typing experience because each one takes some considerable time to execute](https://github.com/ckeditor/ckeditor5-vue/issues/42). 

If you want a synchronous reaction to the editor data changes, attach your own listener on `@ready`:

```js
editor.model.document.on( 'change:data', () => {
    const data = editor.getData();
    console.log( data );
} );
```

It will fire every time something changes in the editor without any delay.

**Comment by darylteo** [comment]
I found a better workaround thanks to @oleq.

```
// @ready event handler
private async onEditorReady(editor: any) {
    this.editorInstance = editor;
}

 private forceUpdateContent() {
    if (!this.editorInstance) {
        return;
    }

    this.model.content = this.editorInstance.getData();
}
```

This allows good performance while typing and fixes dirty read by letting me force read before performing critical actions.

Thanks again1

**Comment by darylteo** [comment]
Actually I'll leave it open in case you want to come up with a better solution to this.

**Comment by LZQCN** [comment]
Combining `editor.GetData` with two-way binding will result in huge performance overhead. For best performance, you should manually call `editor.getData` when you need the result.

**Comment by oleq** [comment]
Closed in #150.
